SYCL/ESIMD/lsc/lsc_predicate.cpp

```diff
@@ -0,0 +1,200 @@
+            auto offset = id[0] * SIMDSize * sizeof(int);
+            auto offsets = simd<uint32_t, SIMDSize>(id * SIMDSize * sizeof(int),
+                                                    sizeof(int));
+            auto pred_enable = simd_mask<1>(1);
+            auto pred_disable = simd_mask<1>(0);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_0,
+                                                                   offset);
+            auto data_0 =
+                lsc_block_load<int, SIMDSize>(access_0, offset, pred_enable);
+            lsc_block_store<int, SIMDSize>(access_0, offset, data_0 * 2,
+                                           pred_enable);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_1,
+                                                                   offset);
+            auto data_1 =
+                lsc_block_load<int, SIMDSize>(access_1, offset, pred_disable);
+            lsc_block_store<int, SIMDSize>(access_1, offset, data_1 * 2,
+                                           pred_enable);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_2,
+                                                                   offset);
+            auto data_2 =
+                lsc_block_load<int, SIMDSize>(access_2, offset, pred_enable);
+            lsc_block_store<int, SIMDSize>(access_2, offset, data_2 * 2,
+                                           pred_disable);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_3,
+                                                                   offset);
+            auto data_3 =
+                lsc_block_load<int, SIMDSize>(access_3, offset, pred_disable);
+            lsc_block_store<int, SIMDSize>(access_3, offset, data_3 * 2,
+                                           pred_disable);
+          });
+    });
+    q.wait();
+    buf_0.template get_access<access::mode::read_write>();
+    buf_1.template get_access<access::mode::read_write>();
+    buf_2.template get_access<access::mode::read_write>();
+    buf_3.template get_access<access::mode::read_write>();
+  } catch (sycl::exception e) {
+    std::cout << "SYCL exception caught: " << e.what();
+    return 1;
+  }
+
+  auto error = 0;
+  for (auto i = 0; i != size; ++i) {
+    error += vec_0[i] != 2 * i;
+    error += vec_1[i] > 0;
+    error += vec_2[i] != i;
+    error += vec_3[i] != i;
+  }
+  std::cout << "Accessor lsc predicate test ";
+  std::cout << (error != 0 ? "FAILED" : "passed") << std::endl;
+  return error;
+}
+
+int testUSM(queue q) {
+  auto size = size_t{128};
+  auto constexpr SIMDSize = unsigned{4};
```

Comment:
I think this is not a practical SIMD size. 8,16,32 would be more relevant. Or, even better, you can templatize by SIMD size and check all of them - like we usually do in other tests.